app/src/main/java/com/group4sweng/scranplan/UserInfoPrivate.java

```diff
@@ -0,0 +1,269 @@
+        editor.putString(IMAGEURL_KEY, getImageURL());
+        editor.putString(ABOUT_KEY, getAbout());
+        editor.putInt(NUM_OF_RECIPES, (int) getNumRecipes());
+        editor.putInt(CHEF_RATING_KEY, (int) getChefRating());
+        editor.putBoolean("ALLERGY_NUTS_KEY", getPreferences().isAllergy_nuts());
+        editor.putBoolean("ALLERGY_SHELLFISH_KEY", getPreferences().isAllergy_shellfish());
+        editor.putBoolean("ALLERGY_MILK", getPreferences().isAllergy_milk());
+        editor.putBoolean("ALLERGY_SOYA", getPreferences().isAllergy_soya());
+        editor.putBoolean("ALLERGY_EGGS", getPreferences().isAllergy_eggs());
+        editor.apply();
+    }
+    */
+
+    /**
+     * Initiate Singleton User Profile Class from a HashMap of values.
+     * Initiated on login/registering for the first time.
+     *
+     * @param map   - Map of user specific information
+     * @param prefs - Map of dietary preferences
+     */
+    public synchronized static void createInstance(HashMap<String, Object> map, HashMap<String, Object> prefs) throws MultipleSingletonCreationException {
+        if (INSTANCE == null) {
+            INSTANCE = new UserInfoPrivate(map, prefs);
+        } else {
+            throw new MultipleSingletonCreationException("Cannot create multiple instances of Singleton class 'UserInfoPrivate'");
+        }
+    }
+
+    /* TODO implement only if Shared Preferences (local storage) is required.
+    /**
+     * Constructor for initiating Singleton User Info Class from the apps 'Login' Activity context
+     * Initiated when user has already logged in and is returning to the MainActivity screen.
+     *
+     * @param context - The context of the current Activity. Should be 'Login'.
+     */
+    /*
+    public synchronized static void createInstance(Context context) throws InvalidContextException {
+
+        //Checks we can only initiate from the Login.class & an instance of the class hasn't already been created.
+        if (checkContext(CONTEXT_LOGIN, context) && INSTANCE == null) {
+            INSTANCE = new UserInfoPrivate(context);
+        } else if (!checkContext(CONTEXT_LOGIN, context) && INSTANCE == null) {
+            throw new InvalidContextException("Activity context required to create instance of class is: " + CONTEXT_LOGIN);
+        } else {
+            Log.e(TAG, "Instance of singleton class 'UserInfoPrivate' already exists");
+        }
+    }
+    */
+
+    /**
+     * Get an instance of the users info (Singleton class) or throw an error if an instance of the class dosen't exist.
+     * It should exist at runtime.
+     *
+     * @return - Valid instance of the class.
+     */
+    public synchronized static UserInfoPrivate getInstance() { //Thread safe
+        if (INSTANCE == null) {
+            throw new java.lang.RuntimeException("Cannot retrieve an instance of the users information from UserInfoPrivate. Make sure an instance of the class is created before retrieving an instance.");
```

Comment:
Can we talk through this on Thursday because this commit stops me being able to make an account(?)

Reply:
Of course yeah. If you bring your laptop I can go through it.